### DISCUSSION TITLE
Allow insert html content into the editor？

### DISCUSSION THREAD

**Discussion by Card007** [Milkdown/milkdown#842] [Q&A]
I will save the html，when I want to edit it，can I insert it to editor ？

**Comment by DaSchTour** [comment]
Currently working on the same thing. Retrieving the HTML is easy. But setting the editor content from some HTML is quite hard.

**Comment by Saul-Mirone** [comment]
HTML is a supported format of default value.
https://milkdown.dev/interacting-with-editor#setting-default-value

**Comment by ahamelers** [comment]
But there is no html insert equivalent of replaceAll ?

**Comment by daviareias** [comment]
I've managed to make the editor convert my html to markdown like this:

```js

// pagetoedit is just a object that I stored with html
const dom = new DOMParser().parseFromString(pagetoedit.html, 'text/html');
Editor.make()
    .config((ctx) => {
        ctx.set(rootCtx, document.getElementById('editor'));
        ctx.set(defaultValueCtx, { type: 'html', dom: dom.firstElementChild });
    })
    .use(commonmark)
    .create()
```
